File: C++/프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.cpp
#include <vector>

using namespace std;

int solution(int x, int y, int n) {
    vector<int> arrive(y + 1, y + 1);
    arrive[x] = 0;
    
    while(x < y) {
        if(x + n <= y &&  arrive[x] + 1 < arrive[x + n]) arrive[x + n] = arrive[x] + 1;
        if(x * 2 <= y &&  arrive[x] + 1 < arrive[x * 2]) arrive[x * 2] = arrive[x] + 1;
        if(x * 3 <= y &&  arrive[x] + 1 < arrive[x * 3]) arrive[x * 3] = arrive[x] + 1;
        x++;
    }
    
    return arrive[y] == y + 1 ? -1 : arrive[y];
}

File: Python3/프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.py
def solution(x, y, n):
    count = [y + 1] * (y + 1)
    count[x] = 0
    if x > y:
        return -1
    
    for i in range(x, y):
        if count[i] == y + 1:
            continue
        if i + n <= y:
            count[i + n] = min(count[i + n], count[i] + 1)
        if i * 2 <= y:
            count[i * 2] = min(count[i * 2], count[i] + 1)
        if i * 3 <= y:
            count[i * 3] = min(count[i * 3], count[i] + 1)
    
    
    return count[y] if count[y] != y + 1 else -1